chromadb/test/ef/test_multimodal_ef.py

```diff
@@ -0,0 +1,150 @@
+    image_embeddings = default_ef(images)
+
+    document_ids = [str(i) for i in range(n_examples, 2 * n_examples)]
+    documents = [random_document() for _ in range(n_examples)]
+    document_embeddings = default_ef(documents)
+
+    # Trying to add a document and an image at the same time should fail
+    with pytest.raises(ValueError):
+        multimodal_collection.add(
+            ids=image_ids[0], documents=documents[0], images=images[0]
+        )
+
+    # Add some documents
+    multimodal_collection.add(ids=document_ids, documents=documents)
+    # Add some images
+    multimodal_collection.add(ids=image_ids, images=images)
+
+    # get() should return all the documents and images
+    # ids corresponding to images should not have documents
+    get_result = multimodal_collection.get(include=["documents"])
+    assert len(get_result["ids"]) == len(document_ids) + len(image_ids)
+    for i, id in enumerate(get_result["ids"]):
+        assert id in document_ids or id in image_ids
+        assert get_result["documents"] is not None
+        if id in document_ids:
+            assert get_result["documents"][i] == documents[document_ids.index(id)]
+        if id in image_ids:
+            assert get_result["documents"][i] is None
+
+    # Generate a random query image
+    query_image = random_image()
+    query_image_embedding = default_ef([query_image])
+
+    image_neighbor_indices, _ = _exact_distances(
+        query_image_embedding, image_embeddings + document_embeddings
+    )
+    # Get the ids of the nearest neighbors
+    nearest_image_neighbor_ids = [
+        image_ids[i] if i < n_examples else document_ids[i % n_examples]
+        for i in image_neighbor_indices[0][:n_query_results]
+    ]
+
+    # Generate a random query document
+    query_document = random_document()
+    query_document_embedding = default_ef([query_document])
+    document_neighbor_indices, _ = _exact_distances(
+        query_document_embedding, image_embeddings + document_embeddings
+    )
+    nearest_document_neighbor_ids = [
+        image_ids[i] if i < n_examples else document_ids[i % n_examples]
+        for i in document_neighbor_indices[0][:n_query_results]
+    ]
+
+    # Querying with both images and documents should fail
+    with pytest.raises(ValueError):
+        multimodal_collection.query(
+            query_images=[query_image], query_texts=[query_document]
+        )
+
+    # Query with images
+    query_result = multimodal_collection.query(
+        query_images=[query_image], n_results=n_query_results, include=["documents"]
+    )
+
+    assert query_result["ids"][0] == nearest_image_neighbor_ids
+
+    # Query with documents
+    query_result = multimodal_collection.query(
+        query_texts=[query_document], n_results=n_query_results, include=["documents"]
+    )
+
+    assert query_result["ids"][0] == nearest_document_neighbor_ids
+
+
+@pytest.mark.xfail
+def test_multimodal_update_with_image(
```

Comment:
can we just add a test documenting the current behavior

Reply:
IMO this is the best way to document 1. that this is the current behavior and 2. that it is not the desired behavior. I will open an issue which references this xfail test to update the behavior, or decide that it's desirable.